### ISSUE TITLE
Unable to save _some_ m2m fields

### ISSUE THREAD

**Issue by reddo** [directus/app#2797]
Directus API version 8.6.2
PHP version 7.3
MySQL version 10.2.27-MariaDB-cll-lve
Apache version 2.2.34

I have a collection with a bunch of many-to-many fields. All of the many-to-many fields have exactly the same configuration. I edit the collection, add values to the m2m fields and some of them throw the following error: 
`Error: Attempting to write an array as the value for field "family_features" in "locations_family_features"`
The payload looks right too:
```
{
  "coordinates": {
    "lat": 46.7713885,
    "lng": 23.5917959
  },
  "cuisine": [
    {
      "cuisine": {
        "id": 2
      }
    },
    {
      "cuisine": {
        "id": 3
      }
    }
  ],
  "family_features": [
    {
      "family_features": {
        "id": 2
      }
    },
    {
      "family_features": {
        "id": 6
      }
    }
  ],
  "features": [
    {
      "features": {
        "id": 6
      }
    }
  ]
}
```
The `features` and `family_features` throw the error. All of the junction collections were auto-generated and look exactly the same to me.

**Comment by reddo** [comment]
I went through the database and I couldn't find any difference between the 2 entities and their relation (collections, junction collections, and fields are all basically the same). I am stumped.

I even deleted and recreated the fields and junction collections. Any ideas as to what else I could try?

**Comment by rijkvanzanten** [comment]
What are the fields in the current, junction, and related fields?

**Comment by reddo** [comment]
@rijkvanzanten, I'm not sure which collections' field you need, but here's what I got that's related
Main collections' m2m field:
![image](https://user-images.githubusercontent.com/1180060/81850365-8d67a380-9560-11ea-9333-4121891c7759.png)
junction collection:
![image](https://user-images.githubusercontent.com/1180060/81850482-b38d4380-9560-11ea-86ea-83d53172efce.png)
Related collection:
![image](https://user-images.githubusercontent.com/1180060/81850529-ca339a80-9560-11ea-8be8-240011f891f9.png)

**Comment by rijkvanzanten** [comment]
Thanks! That's exactly what I was looking for. I'm wondering if it has something to do with the fact that the name of the o2m is the same as the name in the junction table 🤔 That shouldn't cause it to break of course, but that's one explanation why some m2ms fail, but others don't.

**Comment by reddo** [comment]
It's the same for the rest of the m2m fields too. I can try and use different names.

**Comment by reddo** [comment]
@rijkvanzanten, using different names for the junction collection fields seems to work. Now I have to redo the other fields. I'm curious as to why it worked well with the other fields though.

**Comment by rijkvanzanten** [comment]
Yeah me too! I'm starting to suspect that it might be the API mishandling it, as the payload in the OP looks correct to me 🤔
